lib/sepia/src/util.js

```diff
@@ -129,10 +130,7 @@ function substituteWithOpaqueKeys(text) {
   var substitutions = globalOptions.substitutions;
   for (var i=0; i<substitutions.length; i++) {
     var subst = substitutions[i];
-    var actualValue = subst.actualValueFn();
-    while (text.indexOf(actualValue) > -1) {
-      text = text.replace(actualValue, subst.opaqueKey);
-    }
+    text = text.replace(new RegExp(escapeRegex(subst.actualValueFn()), 'g'), subst.opaqueKey);
```

Comment:
doing some performance testing and looks like:
text.split(key).join(value)
might be even faster and this way don't need to bring in another external library 

benchBokor node --version
v6.10.0
benchBokor node bench.js
RegExpReplace x 163,879 ops/sec ±0.57% (91 runs sampled)
SplitJoin x 291,085 ops/sec ±0.82% (92 runs sampled)
Loop x 2,596 ops/sec ±0.40% (95 runs sampled)
RegExpReplaceEscape x 161,572 ops/sec ±0.39% (96 runs sampled)
RegExpReplaceControlRun x 165,887 ops/sec ±0.39% (96 runs sampled)
Fastest is SplitJoin

https://gist.github.com/jimmyeisenhauer/46b1fd875d789daebbff29c22b601bd1